### PR TITLE
[ws_pcan_js]  Add Debug Error Disclosure

### PR DESCRIPTION
**Pull request by eakirtas**

### CODE REVIEW

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/debug_error_disclosure.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// * This Script analyze incoming websocket messages for error messages
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/debug_error_disclosure.js]: "analyze" > "analyzes"

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/debug_error_disclosure.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            matches.forEach(function(evidence){
+                helper.newAlert()
+                    .setName("Information Disclosure - Debug Error Messages via WebSocket (script)")
+                    .setDescription("The response appeared to contain common error messages returned"
+                                    + " by platforms such as ASP.NET, and Web-servers such as IIS and Apache. You can configure"
+                                    + " the list of common debug messages.")
+                    .setSolution("Disable debugging messages before pushing to production.")
+                    .setParam(counter++)
+                    .setEvidence(evidence)
+                    .setCweIdm(200) //Disable debugging messages before pushing to production.
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/debug_error_disclosure.js]: This comment doesn't seem right, I think  you want "CWE-200: Information Exposure" or something like that.

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/debug_error_disclosure.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    debug_messages.forEach(function(pattern){
+        if((matches = message.match(pattern)) != null){
+            matches.forEach(function(evidence){
+                helper.newAlert()
+                    .setName("Information Disclosure - Debug Error Messages via WebSocket (script)")
+                    .setDescription("The response appeared to contain common error messages returned"
+                                    + " by platforms such as ASP.NET, and Web-servers such as IIS and Apache. You can configure"
+                                    + " the list of common debug messages.")
+                    .setSolution("Disable debugging messages before pushing to production.")
+                    .setParam(counter++)
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/debug_error_disclosure.js]: I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/debug_error_disclosure.js (hunk 1 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/debug_error_disclosure.js]: Fixed in 9fb31b841

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/debug_error_disclosure.js (hunk 3 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/debug_error_disclosure.js]: Fixed in d2e4800f6

**Pull request closed without merging**